pkg/backup/backup_compaction_test.go

```diff
@@ -6,67 +6,94 @@
 package backup
 
 import (
+	"context"
 	"fmt"
 	"net/url"
 	"strings"
 	"testing"
 
+	"github.com/cockroachdb/cockroach/pkg/backup/backupinfo"
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/stretchr/testify/require"
 )
 
 func TestBackupCompaction(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
+	ctx := context.Background()
 	tempDir, tempDirCleanup := testutils.TempDir(t)
 	defer tempDirCleanup()
+	st := cluster.MakeTestingClusterSettings()
+	backupinfo.WriteMetadataWithExternalSSTsEnabled.Override(ctx, &st.SV, true)
 	_, db, cleanupDB := backupRestoreTestSetupEmpty(
-		t, singleNode, tempDir, InitManualReplication, base.TestClusterArgs{},
+		t, singleNode, tempDir, InitManualReplication, base.TestClusterArgs{
+			ServerArgs: base.TestServerArgs{
+				Settings: st,
+			},
+		},
 	)
 	defer cleanupDB()
-	fullBackupStmt := "BACKUP INTO 'nodelocal://1/backup'"
 
+	compactionBuiltin := "SELECT crdb_internal.backup_compaction(ARRAY['nodelocal://1/backup'], 'LATEST', ''::BYTES, '%s'::DECIMAL, '%s'::DECIMAL)"
 	t.Run("compaction creates a new backup", func(t *testing.T) {
```

Comment:
this subtest merely asserts that a compaction happened, right? Could this assertion get added to the basic operations subtest, then we can delete this first subtest?